### PR TITLE
[IMP] hr_recruitment: improve applicantion deletion functionality

### PR DESCRIPTION
**Pull request by ravb-odoo**
This PR enhances the functionality of applicantion deletion by ensuring that only applicantions without associated employee records can be deleted. Attempting to delete an applicantion with an associated employee will now result in an error.

task-3660854

### CODE REVIEW

File: addons/hr_recruitment/models/hr_applicant.py [hunk 1]
```diff
@@ -361,6 +361,12 @@ def write(self, vals):
             self._update_employee_from_applicant()
         return res
 
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_linked_employee(self):
+        for applicant in self:
+            if applicant.emp_id:
+                raise UserError(_("Employee already linked with application , you want be deleted this application , instead you archive the record"))
```
Comment (ppr-odoo) [inline addons/hr_recruitment/models/hr_applicant.py:368]:
```suggestion
      if any(applicant.emp_id for applicant in self):
          raise UserError(_('Employee already linked with application, you want be deleted this application, instead you archive the record."'))
```

Comment (ravb-odoo) [reply addons/hr_recruitment/models/hr_applicant.py]: Hello @ppr-odoo ,
Changes are Done.
Thanks

File: addons/hr_recruitment/models/hr_applicant.py [hunk 2]
```diff
@@ -361,6 +361,11 @@ def write(self, vals):
             self._update_employee_from_applicant()
         return res
 
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_linked_employee(self):
+        if any(applicant.emp_id for applicant in self):
```
Comment (Bertrand2) [inline addons/hr_recruitment/models/hr_applicant.py:366]: Accessing `self.emp_id` will return a recordset of all employees from the applicants in self, doing exactly the same thing
```suggestion
        if self.emp_id:
```

**Review by Bertrand2 — CHANGES_REQUESTED** [review]

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**